src/oas2/transformers/request.ts

```diff
@@ -11,21 +11,31 @@ import {
 } from './params';
 
 export function translateToRequest(parameters: Parameter[], consumes: string[]): IHttpOperationRequest {
-  return parameters.reduce((request: IHttpOperationRequest, parameter: Parameter) => {
-    if (isBodyParameter(parameter)) {
-      request.body = translateToBodyParameter(parameter, consumes);
-    } else if (isFormDataParameter(parameter)) {
-      request.body = translateFromFormDataParameter(parameter, request.body, consumes);
-    } else if (isQueryParameter(parameter)) {
-      const queryParameter = translateToQueryParameter(parameter);
-      request.query = (request.query || []).concat(queryParameter);
-    } else if (isPathParameter(parameter)) {
-      const pathParameter = translateToPathParameter(parameter);
-      request.path = (request.path || []).concat(pathParameter);
-    } else if (isHeaderParameter(parameter)) {
-      const headerParameter = translateToHeaderParam(parameter);
-      request.headers = (request.headers || []).concat(headerParameter);
-    }
-    return request;
-  }, {});
+  const bodyParameters = parameters.filter(parameter => isBodyParameter(parameter)) as BodyParameter[];
+  const formDataParameters = parameters.filter(parameter => isFormDataParameter(parameter)) as FormDataParameter[];
+  const request: IHttpOperationRequest = {};
+
+  // if 'body' and 'form data' defined prefer 'body'
```

Comment:
Yeah I think this is a good choice. The spec does not really mandate that a parameter with `formData` excludes `body` and viceversa.